### PR TITLE
fix: limit search term to 100 characters

### LINKED ISSUE #3429: Prevent searches with a query of more than 100 characters in length
https://github.com/mozilla/addons-server/issues/6604 will introduce a limit to the `q` parameter in the search API. When called with a query longer than 100 characters, it will answer with a 400 Bad Request. 

The frontend should prevent the user from making such a request by limiting the user input (and/or be prepared to handle the potential 400 Bad Request response)

### PR DESCRIPTION
**Pull request by tsl143**
Fixes #3429

### CODE REVIEW

File: tests/unit/amo/components/TestSearchForm.js [hunk 1]
```diff
@@ -271,6 +271,13 @@ describe(__filename, () => {
 
       wrapper.find('input').simulate('change', createFakeChangeEvent(undefined));
       expect(wrapper.state('searchValue')).toEqual('');
+
+      // Tests for allowing only 100 characters as input
+      wrapper.find('input').simulate('change', createFakeChangeEvent('asdfghjklpasdfghjklpasdfghjklpasdfghjklpasdfghjklpasdfghjklpasdfghjklpasdfghjklpasdfghjklp98765432111'));
```
Comment (willdurand) [inline tests/unit/amo/components/TestSearchForm.js]: haha maybe we could use [`String.repeat`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/repeat):

```
'c'.repeat(100)
```

File: src/amo/components/SearchForm/index.js [hunk 2]
```diff
@@ -125,7 +125,10 @@ export class SearchFormBase extends React.Component {
   }
 
   handleSearchChange = (event) => {
-    this.setState({ searchValue: event.target.value });
+    const searchValue = event.target.value || '';
+    if (searchValue.length <= SEARCH_TERM_MAX_LENGTH) {
```
Comment (willdurand) [inline src/amo/components/SearchForm/index.js]: We could maybe `trim()` before `.length`?

File: src/amo/components/SearchForm/index.js [hunk 3]
```diff
@@ -160,8 +163,9 @@ export class SearchFormBase extends React.Component {
         but filters have not changed.`);
       return;
     }
-
-    this.dispatchAutocompleteStart({ filters });
+    if (value.length <= SEARCH_TERM_MAX_LENGTH) {
```
Comment (willdurand) [inline src/amo/components/SearchForm/index.js]: Maybe this check could happen earlier?

**Comment by tsl143** [comment]
@willdurand `t` instead of `c` coz its Trishul :P

**Comment by tsl143** [comment]
@willdurand anything more to do on this?

File: src/amo/components/SearchForm/index.js [hunk 4]
```diff
@@ -125,7 +125,10 @@ export class SearchFormBase extends React.Component {
   }
 
   handleSearchChange = (event) => {
-    this.setState({ searchValue: event.target.value });
+    const searchValue = event.target.value || '';
```
Comment (willdurand) [inline src/amo/components/SearchForm/index.js]: Can you add a test case to cover this condition? I think you want to test the case where `event.target.value` is not set or something, and ensure it fallbacks to empty string, avoiding any further issue.

**Comment by willdurand** [comment]
thanks @tsl143!

**Pull request merged**